contracts/basset_vault/src/tests/mod.rs

```diff
@@ -339,15 +345,27 @@ impl WasmMockQuerier {
 
     pub fn with_loan(
         &mut self,
-        borrowers_info: &[(&String, &[(&String, &AnchorBassetCustodyBorrowerInfo)])],
+        borrowers_info: &[(&String, &[(&String, &AnchorMarketBorrowerInfo)])],
     ) {
         self.borrowers_info = array_to_hashmap(borrowers_info);
     }
 
-    pub fn with_wasm_query_response(&mut self, contract_responses_map: &[(&String, &Binary)]) {
-        let mut result_map: HashMap<String, Binary> = HashMap::new();
-        for (contract_addr, response) in contract_responses_map.iter() {
-            result_map.insert((**contract_addr).clone(), (**response).clone());
+    pub fn with_wasm_query_response(
+        &mut self,
+        contract_responses_map: &[(&String, &Binary, &Binary)],
+    ) {
+        let mut result_map: HashMap<String, HashMap<Binary, Binary>> = HashMap::new();
+        for (contract_addr, request, response) in contract_responses_map.iter() {
+            match result_map.get_mut(*contract_addr) {
+                Some(by_contract_responses) => {
+                    by_contract_responses.insert((**request).clone(), (**response).clone());
+                }
+                None => {
+                    let mut by_contract_responses = HashMap::new();
+                    by_contract_responses.insert((**request).clone(), (**response).clone());
+                    result_map.insert((**contract_addr).clone(), by_contract_responses);
+                }
+            }
```

Comment:
Check out this API https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry. It looks like it can be rewritten a lot easier
```rust
result_map
    .entry(contract_addr)
    .or_default()
    .insert((**request).clone(), (**response).clone())
```

Reply:
indeed, thanks!